lib/aurora_bootstrapper/notifier.rb

```diff
@@ -12,7 +14,27 @@ def export_date
     
     # ENV is string to string dictionary
     def export_date_override
-      DateTime.now.strftime("%Y-%m-%d") if ENV.key?('EXPORT_DATE_OVERRIDE')
+      return nil unless ENV.key?('EXPORT_DATE_OVERRIDE')
+
+      today = Date.today
+      days_ago = (0..30).find do | days_ago |
+        exists_export? date: today-days_ago
+      end
+      
+      converted_days_ago = (days_ago.nil? || days_ago.to_i == 0) ? 1 : days_ago.to_i
+      ( today - converted_days_ago + 1).strftime( DATE_FORMAT )
+    end
+
+    def exists_export?( date: )
+      prefix = [ bucket_path, date.strftime( DATE_FORMAT ) ].join( '/' )
+
+      found_object = client.list_objects_v2({ 
+        bucket: bucket,
+        prefix: prefix }).contents.find do | object | 
+          object.key.include? "DONE"
+        end
+
+      !found_object.nil?
```

Comment:
this is no longer strictly necessary - `.find` will either return a truthy value from the array or a falsey `nil`.

Reply:
Cool